frontend/app/[team]/apps/[app]/environments/[environment]/[[...path]]/page.tsx

```diff
@@ -735,6 +755,156 @@ export default function EnvironmentPath({
     </SplitButton>
   )
 
+  const handleChangesReveal = () => {
+    const changes = getChanges();
+    setAllChanges(changes);
+    setIsChangesModalOpen(true); 
+  };
+
+  const getChanges = (): Record<string, SecretChange> => {
+    const changes: Record<string, SecretChange> = {};
+  
+    clientSecrets.forEach((updatedSecret) => {
+      const originalSecret = serverSecrets.find(
+        (serverSecrets) => serverSecrets.id === updatedSecret.id
+      );
+  
+      if (!originalSecret) {
+        changes[updatedSecret.key] = {
+          type: "Added",
+          secretName: updatedSecret.key,
+          key: {
+            old: "",
+            new: updatedSecret.key,
+          },
+          value: {
+            old: "",
+            new: updatedSecret.value,
+          },
+          comment: {
+            old: "",
+            new: updatedSecret.comment,
+          },
+          tags: {
+            old: [],
+            new: updatedSecret.tags,
+          },
+        };
+        return;
+      }
+  
+      const secretChanges: Partial<SecretChange> = {
+        type: "Modified",
+        secretName: originalSecret.key,
+      };
+  
+      if (originalSecret.key !== updatedSecret.key) {
+        secretChanges.key = {
+          old: originalSecret.key,
+          new: updatedSecret.key,
+        };
+      }
+  
+      if (originalSecret.value !== updatedSecret.value) {
+        secretChanges.value = {
+          old: originalSecret.value,
+          new: updatedSecret.value,
+        };
+      }
+  
+      if (originalSecret.comment !== updatedSecret.comment) {
+        secretChanges.comment = {
+          old: originalSecret.comment || "",
+          new: updatedSecret.comment,
+        };
+      }
+  
+      if (!arraysEqual(originalSecret.tags, updatedSecret.tags)) {
+        secretChanges.tags = {
+          old: originalSecret.tags,
+          new: updatedSecret.tags,
+        };
+      }
+  
+      if (
+        secretChanges.key ||
+        secretChanges.value ||
+        secretChanges.comment ||
+        secretChanges.tags
+      ) {
+        changes[updatedSecret.key] = secretChanges as SecretChange;
+      }
+    });
+  
+    return changes;
+  };
+
+  const limitString = (str: string) => { 
+    if (str.length <=50) return str
+    return str.slice(0, 50) + '...';
+  } 
+  
+  const DisplayChanges = ({ change }: any) => {
+    const removedTags = change.tags?.old?.filter(tag => 
+      !change.tags?.new?.some(newTag => newTag.id === tag.id)
+    ) || [];
+  
+    const addedTags = change.tags?.new?.filter(tag => 
+      !change.tags?.old?.some(oldTag => oldTag.id === tag.id)
+    ) || [];
```

Comment:
```suggestion
    const removedTags =
      change.tags?.old?.filter(
        (tag: SecretTagType) =>
          !change.tags?.new?.some((newTag: SecretTagType) => newTag.id === tag.id)
      ) || []

    const addedTags =
      change.tags?.new?.filter(
        (tag: SecretTagType) =>
          !change.tags?.old?.some((oldTag: SecretTagType) => oldTag.id === tag.id)
      ) || []
```

Add the `SecretTagType` annotation here